### PR TITLE
Wrote security configuration for counter part

### PR DESCRIPTION
**Pull request by justin-obrien-smoothstack**
No unit tests yet, but tested manually and everything seems to work. Includes #6, so after that gets merged there will be fewer changes to review here.

### CODE REVIEW

File: src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		return authenticationManager.authenticate(authenticationToken);
+	}
+
+    @Override
+	protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain,
+			Authentication authResult) throws IOException, ServletException {
+		UserDetailsClass userDetails = (UserDetailsClass) authResult.getPrincipal();
+		String token = JWT.create().withSubject(userDetails.getUsername())
+				.withExpiresAt(new Date(System.currentTimeMillis() + JwtProperties.EXPIRATION_TIME))
+				.sign(HMAC512(JwtProperties.SECRET.getBytes()));
```
Comment (trevorhuis-smoothstack) [inline src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthenticationFilter.java:62]: Akbar said we need to change this from HMAC512 to SHA256 in the other group's demo. Are you familiar with how that works?

File: src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	}
+
+	private Authentication getUsernamePasswordAuthentication(HttpServletRequest request)
+			throws UsernameNotFoundException {
+		String token = request.getHeader(JwtProperties.HEADER_STRING), username;
+		UsernamePasswordAuthenticationToken authToken;
+		User user;
+		UserDetailsClass userDetails;
+		if (token != null) {
+			username = JWT.require(HMAC512(JwtProperties.SECRET.getBytes())).build()
```
Comment (trevorhuis-smoothstack) [inline src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthorizationFilter.java:56]: Similar comment as above pertaining to switching HMAC512 to SHA256.

**Review by trevorhuis-smoothstack — APPROVED** [review]
Looks like we have the same setup so we are all on the same page. I see you created a folder for just counter security. I will do the same with agent.

File: src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthenticationFilter.java (hunk 1 above)
Comment (justin-obrien-smoothstack) [reply src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthenticationFilter.java:62]: I'm not sure, but hopefully it'll just be something simple like importing a SHA256 method and using it in place of HMAC512, like we were talking about in our meeting today.

**Pull request merged**